### ISSUE TITLE
auto import always choose the type declaration in the folder that not top level

### ISSUE THREAD

**Issue by Kingwl** [Microsoft/TypeScript#24742]
**TypeScript Version:**  3.0.0-dev.201xxxxx

**Search Terms:** 

**Code**

```ts
// node_module/xxx/lib/index.d.ts
export declare a: number;

// src/a.ts
a/*here*/

```

**Expected behavior:**

insert the code: `import { a } from 'xxx'`

**Actual behavior:**

insert the code: `import { a } from 'xxx/lib'`

**Playground Link:**  

**Related Issues:**

**Comment by mhegazy** [comment]
Following this repro, i am not even getting completions for `a`.

**Comment by andy-ms** [comment]
@Kingwl Assuming that the problem here was that there was a "main" field in the `package.json` referencing `lib/index.js`, but no `typings` field. If so, should be fixed by #24774.
